Extensions/TileMap/pixi-tilemap-helper.js

```diff
@@ -0,0 +1,261 @@
+      const columnMultiplier = Math.floor((frame - 1) % columns)
+      const rowMultiplier = Math.floor((frame - 1) / columns)
+      const x = (margin + (columnMultiplier * (tileWidth + spacing)))
+      const y = margin + (rowMultiplier * (tileHeight + spacing))
+
+      try {
+        const rect = new PIXI.Rectangle(x, y, tileWidth, tileHeight)
+        const texture = new PIXI.Texture(tex, rect)
+        texture.baseTexture.scaleMode = PIXI.SCALE_MODES.NEAREST
+        texture.cacheAsBitmap = true
+
+        return texture
+      } catch (error) {
+        console.log(rect);
+        console.error(
+          'Error occured while trying to create PIXI.Texture!',
+          error
+        )
+        return null
+      }
+    })
+    const newTileset = {
+      width: tex.width,
+      height: tex.height,
+      tileWidth,
+      tileHeight,
+      texture: tex,
+      textureCache,
+      layers: tiledData.layers,
+      tiles,
+      tilecount,
+      // these scales are used as multipliers on some absolute pixel values from tiled that might be on a resized atlas
+      // they help to adopt to changed dimentions when rendering in updatePIXITileMap
+      margin,
+      spacing,
+    }
+    onLoad(newTileset)
+    loadedTileSets[requestedTileSetId] = newTileset
+  }
+  exports.createTileSetResource = createTileSetResource
+
+  /**
+   * Decodes the layer data, which tiled can sometimes store as a compressed base64 string
+   * https://doc.mapeditor.org/en/stable/reference/tmx-map-format/#data
+   */
+  const decodeBase64 = (layer, pako) => {
+    const { data, compression } = layer
+    var index = 4,
+      arr = [],
+      step1 = atob(data)
+        .split('')
+        .map(function (x) {
+          return x.charCodeAt(0)
+        })
+    try {
+      const decodeString = (str, index) =>
+        (str.charCodeAt(index) +
+          (str.charCodeAt(index + 1) << 8) +
+          (str.charCodeAt(index + 2) << 16) +
+          (str.charCodeAt(index + 3) << 24)) >>>
+        0
+      const decodeArray = (arr, index) =>
+        (arr[index] +
+          (arr[index + 1] << 8) +
+          (arr[index + 2] << 16) +
+          (arr[index + 3] << 24)) >>>
+        0
+
+      if (compression === 'zlib') {
+        var binData = new Uint8Array(step1)
+        step1 = pako.inflate(binData)
+        while (index <= step1.length) {
+          arr.push(decodeArray(step1, index - 4))
+          index += 4
+        }
+      } else {
+        while (index <= step1.length) {
+          arr.push(decodeString(step1, index - 4))
+          index += 4
+        }
+      }
+      return arr
+    } catch (error) {
+      console.error(
+        'Failed to decompress and unzip base64 layer.data string',
+        error
+      )
+      return null
+    }
+  }
+
+  /**
+   * Re-renders the tilemap whenever its rendering settings have been changed
+   */
+  exports.updatePIXITileMap = (tileMap, tileSet, render, layerIndex, pako) => {
```

Comment:
Is `render` the proper name for this? It sounds like a verb, and a verb would designate a function. I think it was renamed in the rest of the codebase? At least "render" is way too generic, and should be a name for a function, not a setting.

Reply:
good catch, updated the name now